Python/optimizer.c

```diff
@@ -611,6 +615,10 @@ translate_bytecode_to_trace(
                                 Py_FatalError("garbled expansion");
                         }
                         ADD_TO_TRACE(expansion->uops[i].uop, oparg, operand);
+                        if (expansion->uops[i].uop == _PUSH_FRAME) {
+                            ADD_TO_TRACE(SAVE_IP, 0, 0);
+                            goto done;
```

Comment:
Maybe assert that this is the last uop in the expansion (since we're bailing out of the inner loop)?

Reply:
Good idea.
```
assert(i + 1 == nuops);
```